code/datums/components/ntnet_interface.dm

```diff
@@ -1,66 +1,136 @@
 /datum/proc/ntnet_receive(datum/netdata/data)
-	return
+	SEND_SIGNAL(src, COMSIG_COMPONENT_NTNET_RECEIVE, data)
 
-/datum/proc/ntnet_receive_broadcast(datum/netdata/data)
-	return
 
-/datum/proc/ntnet_send(datum/netdata/data, netid)
+// helper function.  So you don't have to get the component
+/datum/proc/ntnet_send(datum/netdata/data)
 	var/datum/component/ntnet_interface/NIC = GetComponent(/datum/component/ntnet_interface)
 	if(!NIC)
 		return FALSE
-	return NIC.__network_send(data, netid)
+	data.sender_id = NIC.hardware_id
+	data.network_id = NIC.network.network_id
+	return SSnetworks.transmit(data)
+
 
 /datum/component/ntnet_interface
-	var/hardware_id			//text. this is the true ID. do not change this. stuff like ID forgery can be done manually.
-	var/network_name = ""			//text
-	var/list/networks_connected_by_id = list()		//id = datum/ntnet
-	var/differentiate_broadcast = TRUE				//If false, broadcasts go to ntnet_receive. NOT RECOMMENDED.
+	var/hardware_id = null				// text. this is the true ID. do not change this. stuff like ID forgery can be done manually.
+	var/id_tag = null  					// named tag for looking up on mapping objects
+	var/datum/ntnet/network = null		// network we are on, we MUST be on a network or there is no point in this component
+	var/list/registered_sockets = null	// list of ports opened up on devices
+	var/list/network_alias = list() 		// if we live in more than one network branch TODO
+
+/datum/component/ntnet_interface/Initialize(network_name, network_tag = null)
+	if(!network_name)
+		stack_trace("Bad network '[network_name]' for '[parent]', going to limbo it")
+		network_name = NETWORK_LIMBO
+
+	if(network_tag != null && text2num(network_tag) == text2num(num2text(network_tag)))
+		// numbers are not allowed as lookups for interfaces
+		stack_trace("Tag cannot be a number?  '[network_name]' for '[parent]', going to limbo it")
+		network_tag = "BADTAG_" + network_tag
 
-/datum/component/ntnet_interface/Initialize(force_name = "NTNet Device", autoconnect_station_network = TRUE)			//Don't force ID unless you know what you're doing!
 	hardware_id = "[SSnetworks.get_next_HID()]"
-	network_name = force_name
-	if(!SSnetworks.register_interface(src))
-		. = COMPONENT_INCOMPATIBLE
-		CRASH("Unable to register NTNet interface. Interface deleted.")
-	if(autoconnect_station_network)
-		register_connection(SSnetworks.station_network)
+	id_tag = network_tag
+	SSnetworks.interfaces_by_hardware_id[hardware_id] = src
+	registered_sockets = list()
+
+	join_network(network_name)
+
+
+// Port connection system
+// The basic idea is that two or more objects share a list and transfer data between the list
+// The list keeps a flag called "_updated", if that flag is set to "true" then something was
+// changed.  Now I COULD send a signal, but that would require the parent object to be shoved
+// in datum/netlink.  I am trying my best to not have hard references in any of these data
+// objects
```

Comment:
This is not a valid dmdoc comment, please re-visit our [code documentation standards](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md).

Reply:
Really sorry about this, completely went over my head on getting the documentation properly written.  Just started writing it.  Just finished the subsystem documentation on the latest push if you could review it.  Going to start on some of the classes I added now.